src/advanced-settings/data/api.js

```diff
@@ -26,7 +38,13 @@ export async function getCourseAdvancedSettings(courseId) {
 export async function updateCourseAdvancedSettings(courseId, settings) {
   const { data } = await getAuthenticatedHttpClient()
     .patch(`${getCourseAdvancedSettingsApiUrl(courseId)}`, convertObjectToSnakeCase(settings));
-  return camelCaseObject(data);
+  const objectFormatted = camelCaseObject(data);
+  return modifyObjectKeys(objectFormatted, (key) => {
+    if (objectFormatted[key]?.value) {
+      objectFormatted[key].value = snakeCaseObject(objectFormatted[key].value);
+    }
+    return key;
+  });
```

Comment:
This doesn't seem like a good use of `modifyObjectKeys`. It's just iterating over the object values, right? The keys are not actually being modified.

I think something like this is better?

```suggestion
  Object.keys(objectFormatted).forEach((key) => {
    if (objectFormatted[key]?.value) {
      objectFormatted[key].value = snakeCaseObject(objectFormatted[key].value);
    }
  });
  return objectFormatted;
```

Also, code like this can get very confusing to follow. It is way easier for other devs if you include comments that give examples of what the data looks like at each step. Like:

```ts
// we start with: { camelCaseField: { value: { shouldBeSnakeCase: '123' } }, ... }
// we want: { camelCaseField: { value: { should_be_snake_case: '123' } }, ... }
```